providers/src/airflow/providers/openlineage/plugins/listener.py

```diff
@@ -139,11 +139,19 @@ def on_running():
                 logical_date=dagrun.logical_date,
             )
 
-            task_uuid = self.adapter.build_task_instance_run_id(
+            if not task_instance.queued_dttm:
+                # This should not happen outside of tests, but if it does, we should log it and continue
+                self.log.warning("Task instance %s has no queued_dttm", task_instance)
+
+            task_uuid = OpenLineageAdapter.build_task_instance_run_id(
                 dag_id=dag.dag_id,
                 task_id=task.task_id,
                 try_number=task_instance.try_number,
                 execution_date=task_instance.execution_date,
+                queued_dttm=task_instance.queued_dttm
+                if task_instance.queued_dttm
+                else task_instance.execution_date,
```

Comment:
Are we testing this particular `if` somewhere?

Reply:
Only instance I've seen this happen is actually in testing. 
There's log over that warns in this case.